### ISSUE TITLE
Incorporate Reconstruction Loss from VAE Arm into HL-NET Loss Func

### ISSUE THREAD

**Issue by fridells51** [noahdrakes/mldl-final#2]
Figure out a way to balance the loss function to incorporate reconstruction loss into the HL-NET.

One idea is to figure out a trainable way to incorporate this reconstruction into the HL-Net model. There should be some way to connect the VAE arm (which acts as a discriminator) to the rest of the model so that it learns to interpret the outputs of the pre-trained VAE.

Another idea is to do a weighted average of the loss function that includes the VAE reconstruction loss output.

Another idea is to augment the data with an "inductive bias" generated from the output of the frozen encoder

**Comment by fridells51** [comment]
Here's where I stand at this point: 

Adding the scalar value of reconstruction loss to the loss function *does* work, it's just not very elegant. Even though the recon loss contribution from the frozen VAE won't change during adaptively training (since VAE weights are fixed). This signal still will help the model because the HL-net shares input with the VAE. It is basically getting feedback and would hopefully addapt to large reconstruction loss contributions. However, we would have to make a hyperparameter choice to balance the recon loss contribution in the HL-net loss function. 

But basically, since the reconstruction loss is more or less a signal, then this could pretty much just be added to the data. However, the more I think about this, the less robust it seems. Adding recon loss to the loss func acts as a fixed pressure on optimization and the HL-Net model must learn to resolve this signal that it is basically blind to until the backward step. If the model has direct access to the VAE's signals as features, then it could potentially learn to over-rely on these features. Because the features are not actual components of the data, this could be viewed as a shortcut or cheating.
